spharpy/sht.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+from pyfar import Signal
+import warnings
+from . import SphericalHarmonicSignal
+from . import SphericalHarmonics
+
+
+def sht(signal, coordinates, n_max, basis_type="real", axis=-2,
+        channel_convention='acn', normalization='n3d', condon_shortley='auto',
+        inverse_method='pseudo_inverse'):
+    """Compute the spherical harmonics transform
+
+    Parameters
+    ----------
+    signal: Signal
+        the signal for which the spherical harmonics transform is computed
+    coordinates: :class:`spharpy.samplings.Coordinates`, :doc:`pf.Coordinates
+        <pyfar:classes/pyfar.coordinates>`
+        Coordinates on which the signal has been sampled
+    n_max: integer
+        Spherical harmonic order
+    basis_type: str
+        Use real or complex valued SH bases ``'real'``, ``'complex'``
+        default is ``'real'``
+    axis: integer
+        Axis along which the SH transform is computed
+    channel_convention : str
+        Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+        (FuMa is only supported up to 3rd order)
+    normalization : str
+        Normalization convention, either ``'n3d'``, ``'maxN'`` or
+        ``'sn3d'``. (maxN is only supported up to 3rd order)
+    condon_shortley : bool or str
+        Flag to indicate if the Condon-Shortley phase term is included
+        (``True``) or not (``False``).
+    inverse_method: str
+        Method for computing the inverse transform:
+        - ‘auto’: use ‘quadrature’ when applicable, otherwise ‘pseudo_inverse’.
+        - ‘quadrature’: compute the inverse via numerical quadrature.
+        - ‘pseudo_inverse’: compute the inverse via a pseudo-inverse approximation.
+
+    Returns
+    ----------
+    SphericalHarmonicSignal
+
+    References
+    ----------
+
+        [#] Rafaely, B. (2015). Fundamentals of Spherical Array Processing,
+            (J. Benesty and W. Kellermann, Eds.) Springer Berlin Heidelberg,
+            2nd ed., 196 pages. doi:10.1007/978-3-319-99561-8
+        [#] Ramani Duraiswami, Dimitry N. Zotkin, and Nail A. Gumerov: "Inter-
+            polation and range extrapolation of HRTFs." IEEE Int. Conf.
+            Acoustics, Speech, and Signal Processing (ICASSP), Montreal,
+            Canada, May 2004, p. 45-48, doi: 10.1109/ICASSP.2004.1326759.
+    """
+
+    if not signal.time.shape[axis] == coordinates.csize:
+        if coordinates.csize not in signal.time.shape:
+            raise ValueError("Signal shape does not match "
+                             "number of coordinates.")
+        else:
+            axis = signal.time.shape.index(coordinates.csize)
+            warnings.warn("Compute spherical harmonics transform along "
+                          f"axis = {axis}.", UserWarning)
```

Comment:
I think we should only raise an error here and not change the user input. But I like the idea of detecting the axis. How about having `axis='auto'` as a default and use

`axis = np.where(np.array(signal.cshape) == Y_inv.shape[1])[0]` 

to detect the axis. If `len(axis) == 1` we proceed if `len(axis) == 0` we raise an error and if `len(axis) > 1` we raise a different errror.